File: android/src/main/java/com/kishannareshpal/expopdf/lib/records/ContentPadding.kt
package com.kishannareshpal.expopdf.lib.records

import expo.modules.kotlin.records.Field
import expo.modules.kotlin.records.Record

class ContentPadding: Record {
  @Field
  val left: Int = 0

  @Field
  val top: Int = 0

  @Field
  val right: Int = 0

  @Field
  val bottom: Int = 0
}

File: ios/lib/ContentPadding.swift
//
//  ContentPadding.swift
//  Pods
//
//  Created by Kishan Jadav on 05/01/2026.
//

import ExpoModulesCore

struct ContentPadding: Record {
  @Field
  var left: Int = 0
  
  @Field
  var top: Int = 0
  
  @Field
  var right: Int = 0
  
  @Field
  var bottom: Int = 0
  
  func toEdgeInset() -> UIEdgeInsets {
    UIEdgeInsets(top: CGFloat(self.top), left: CGFloat(self.left), bottom: CGFloat(self.bottom), right: CGFloat(self.right))
  }
}
